src/Transaction.ts

```diff
@@ -56,7 +65,18 @@ export class Transaction {
       to: contractAddress,
       value: options.value || new BigNumber(0)
     }
-    const ethFees = rawTx.gasLimit.multipliedBy(rawTx.gasPrice)
+    if (options.delegationFees && options.currencyNetworkOfFees) {
+      rawTx.delegationFees = options.delegationFees
+      rawTx.currencyNetworkOfFees = options.currencyNetworkOfFees
+    } else {
+      const metaTransactionFees: MetaTransactionFees = await this.signer.getMetaTxFees(
+        rawTx
+      )
+      rawTx.delegationFees = metaTransactionFees.delegationFees
+      rawTx.currencyNetworkOfFees = metaTransactionFees.currencyNetworkOfFees
+    }
+
+    const ethFees = new BigNumber(rawTx.gasLimit).multipliedBy(rawTx.gasPrice)
     return {
       ethFees: utils.formatToAmountInternal(ethFees, ETH_DECIMALS),
```

Comment:
Here we return the eth fees, so we should here probably also return the delegation fees. 
The same for the prepare methods in transfer and trustline.